src/main/dialogs/basic/blurrable.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * A dialog which is hidden when it loses focus.
+ *
+ * A dialog which is hidden whenever it or a window which is "linked" to it
+ * loses focus. The performed action can be customized by the implementor.
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+
+var fluid = require("infusion"),
+    BrowserWindow = require("electron").BrowserWindow,
+    gpii = fluid.registerNamespace("gpii");
+
+/**
+ * A dialog which is hidden whenever it or a window which is "linked" to it
+ * loses focus AND the newly focused window is not "linked" to the current
+ * window. Each dialog defines the other dialogs to which it is linked
+ * by specifying at least one of their gradeNames in the `linkedWindowsGrades`
+ * array. The "linked to" relation is always reflexive but may not be symmetric.
+ * Note that the action performed when the window (or a linked one) loses focus
+ * can be customized by the implementor.
+ */
+fluid.defaults("gpii.app.blurrable", {
+    gradeNames: ["fluid.component"],
+
+    linkedWindowsGrades: [],
+
+    events: {
+        // All blurrable dialogs share one and the same `onBlur` event so that
+        // they can hide themselves even if they have already lost focus but
+        // were not hidden because a linked window has gained it
+        onBlur: "{app}.events.onBlur"
+    },
+
+    listeners: {
+        "onCreate.initBlurrable": {
+            funcName: "gpii.app.blurrable.initBlurrable",
+            args: ["{that}"]
+        },
+        onBlur: {
+            funcName: "gpii.app.blurrable.onBlur",
+            args: ["{that}"]
+        }
+    },
+
+    invokers: {
+        // The action which should be performed when the window loses focus
+        handleBlur: {
+            funcName: "gpii.app.blurrable.handleBlur",
+            args: ["{that}"]
+        }
+    }
+});
+
+/**
+ * Initializes the component by registering a blur listener for the member
+ * dialog.
+ * @param {Component} that - The `gpii.app.blurrable` instance.
+ */
+gpii.app.blurrable.initBlurrable = function (that) {
+    // Attach the grade names as window parameters
+    that.dialog.gradeNames = that.options.gradeNames;
```

Comment:
Any reason we can't just use them where they stand in that.options?

Reply:
We are using `BrowserWindow.getFocusedWindow()` which returns the currently focused `BrowserWindow` in order to determine if the dialog of the current component should close as a result of a blur somewhere in the component tree. We thought it would be better for the dialog to know about the gradeNames of the component to which belongs instead of to iterate through all components that are dialogs whenever a blur occurs.

Reply:
OK, I see - please leave an explanatory comment explaining what is the scope of this usage. Also, write a placeholder comment in the "dialog" member definition of gpii.app.dialog indicating that these grades may be present. If this usage expands further it may be clearer to just inject the entire component or else implement a dedicated id-based lookup from dialogs to components.